pkg/artifactory/user.go

```diff
@@ -36,7 +37,7 @@ var baseUserSchema = map[string]*schema.Schema{
 	"email": {
 		Type:             schema.TypeString,
 		Required:         true,
-		ValidateDiagFunc: validation.ToDiagFunc(validateIsEmail),
+		ValidateDiagFunc: validation.ToDiagFunc(utils.ValidateIsEmail),
```

Comment:
Instead of upgrading the function, can we just declare `ValidateIsEmail` is of the right function type from the start?

Reply:
Sure we can do that. All the changes here relates to ensuring code compiles and functions correctly. No improvements were made. We can do that in subsequent PRs.